source/tools/testConsistency/dotnetWPF-renderer/WPFRenderer/WPFRenderer/Program.cs

```diff
@@ -0,0 +1,112 @@
+                }
+                while ((currLine != null) && (currLine != ""));
+
+                JObject jsonInput = JObject.Parse(inputSb.ToString());
+                string cardData = jsonInput.GetValue("data").ToString();
+                string cardTemplate = jsonInput.GetValue("template").ToString();
+
+                AdaptiveCardTemplate adaptiveCardTemplate = new AdaptiveCardTemplate(cardTemplate);
+                string adaptiveCardJson = adaptiveCardTemplate.Expand(cardData);
+                AdaptiveCardParseResult parseResult = AdaptiveCard.FromJson(adaptiveCardJson);
+                AdaptiveCardRenderer renderer = new AdaptiveCardRenderer();
+
+                GetResultJsonString(parseResult, renderer, resultJson, warningList).GetAwaiter().GetResult();
+            }
+
+            catch (Newtonsoft.Json.JsonReaderException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("The input is not a valid json");
+            }
+            catch (AdaptiveTemplateException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("Unable to bind the data with the template. The supplied card data is not valid");
+            }
+            catch (AdaptiveSerializationException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("The supplied json input is not a valid Adaptive Card");
+
+            }
+            catch (ArgumentNullException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("Unable to capture the image of the card. Ensure that the card template and card data are valid and try again.");
+            }
+            catch (Exception e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("An error was encountered: " + e.Message);
+            }
+
+            finally
+            {
+                resultJson["warnings"] = new JArray(warningList);
+                resultJson["errors"] = new JArray(errorList);
+                Console.WriteLine(resultJson.ToString());
+            }
+        }
+
+        private static async Task GetResultJsonString(AdaptiveCardParseResult parseResult, AdaptiveCardRenderer renderer, JObject resultJson, List<string> warningList)
+        {
+            AdaptiveCard adaptiveCard = parseResult.Card;
+            CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+            RenderedAdaptiveCardImage adaptiveCardImage = await renderer.RenderCardToImageAsync(adaptiveCard, true, 400, cts.Token);
+
+            foreach (var warning in parseResult.Warnings.Union(adaptiveCardImage.Warnings))
+            {
+                warningList.Add(warning.Message);
+            }
+
+            Stream cardImgStream = adaptiveCardImage.ImageStream;
```

Comment:
>Stream cardImgStream = adaptiveCardImage.ImageStream; [](start = 12, length = 53)

Are you potentially leaking any native resources here? Does ImageStream require you to dispose off the stream after using? (in other words does it implement a Dispose() pattern?) If so this would be under a using() statement to automatically invoke stream.Dispose() and release any native handles it might have been holding on to after going out of scope.